### PR TITLE
Add com.adrienplazas.Metronome

### PR DESCRIPTION
**Pull request by Kekun**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.adrienplazas.Metronome.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
+        ],
+        "test-args": [
+            "--socket=x11",
+            "--share=network"
+        ],
```
Comment (hfiguiere) [inline com.adrienplazas.Metronome.json:22]: there is no network on the builder. so if expect to work (like for cargo), it won't.

Comment (hfiguiere) [reply com.adrienplazas.Metronome.json:22]: and there is no X11 either. So 'run-tests' will like fail if you expect this

**Comment by hfiguiere** [comment]
bot, build com.adrienplazas.Metronome

File: com.adrienplazas.Metronome.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        "--device=dri",
+        "--socket=fallback-x11",
+        "--socket=pulseaudio",
+        "--socket=wayland"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
+        ],
```
Comment (hfiguiere) [inline com.adrienplazas.Metronome.json:18]:
```suggestion
```

File: com.adrienplazas.Metronome.json (hunk 1 above)
Comment (hfiguiere) [inline com.adrienplazas.Metronome.json:22]:
```suggestion
```

File: com.adrienplazas.Metronome.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                            "commit" : "82803377c33247265d779af034eceb5949e78354"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "metronome",
+            "buildsystem": "meson",
+            "run-tests": true,
```
Comment (hfiguiere) [inline com.adrienplazas.Metronome.json:79]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
It's probably still missing the cargo dependencies.

See https://github.com/flatpak/flatpak-builder-tools

**Comment by bilelmoussaoui** [comment]
> It's probably still missing the cargo dependencies.
> 
> See https://github.com/flatpak/flatpak-builder-tools

That won't work with cargo + meson. Please run ninja dist & upload the tarball somewhere

**Comment by hfiguiere** [comment]
Good To Know.

**Comment by Kekun** [comment]
Hopefully this time it will be good, thanks for your help! :)

**Comment by hfiguiere** [comment]
bot, build com.adrienplazas.Metronome

**Comment by hfiguiere** [comment]
Failure is unrelated, a problem on the builders.

**Comment by AsciiWolf** [comment]
bot, build com.adrienplazas.Metronome

**Comment by AsciiWolf** [comment]
@bilelmoussaoui It looks like that there is no space left on the "sources-1" worker.

**Comment by hfiguiere** [comment]
> @bilelmoussaoui It looks like that there is no space left on the "sources-1" worker.

It's a more general problem.

**Comment by AsciiWolf** [comment]
> It's a more general problem.

#2450

**Comment by hfiguiere** [comment]
bot, build com.adrienplazas.Metronome

File: com.adrienplazas.Metronome.json [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        "--device=dri",
+        "--socket=fallback-x11",
+        "--socket=pulseaudio",
+        "--socket=wayland"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "env" : {
+            "CARGO_HOME" : "/run/build/metronome/cargo",
+            "RUSTFLAGS" : "-L=/app/lib"
```
Comment (bilelmoussaoui) [inline com.adrienplazas.Metronome.json:20]:
```suggestion
            "CARGO_HOME" : "/run/build/metronome/cargo"
```
this was only needed at times where you needed to build gtk4 yourself and was used to fix a pango linking issue. Can be safely dropped

File: com.adrienplazas.Metronome.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/libadwaita.git",
+                    "tag" : "1.0.0-alpha.2",
+                    "commit" : "f5932ab4250c8e709958c6e75a1a4941a5f0f386"
+                }
+            ],
+            "modules": [
+                {
+                    "name" : "libsass",
+                    "buildsystem" : "meson",
```
Comment (bilelmoussaoui) [inline com.adrienplazas.Metronome.json:62]: `"cleanup": ["*"]` for both libsass & sassc

File: com.adrienplazas.Metronome.json [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        "/lib/girepository-1.0",
+        "/share/gir-1.0",
+        "/bin/sassc"
+    ],
+    "modules": [
+        {
+            "name" : "libadwaita",
+            "buildsystem" : "meson",
+            "config-opts" : [
+                "-Dintrospection=enabled",
```
Comment (bilelmoussaoui) [inline com.adrienplazas.Metronome.json:42]:
```suggestion
                "-Dintrospection=disabled",
```
not needed

**Comment by bilelmoussaoui** [comment]
looks good to me otherwise

File: com.adrienplazas.Metronome.json (hunk 4 above)
Comment (Kekun) [reply com.adrienplazas.Metronome.json:20]: I assume Solanum should be updated then, this is where I copied it from: https://github.com/flathub/org.gnome.Solanum/blob/master/org.gnome.Solanum.json#L21

**Comment by nedrichards** [comment]
bot, build com.adrienplazas.Metronome

File: com.adrienplazas.Metronome.json [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                    "tag" : "1.0.0-alpha.2",
+                    "commit" : "f5932ab4250c8e709958c6e75a1a4941a5f0f386"
+                }
+            ],
+            "modules": [
+                {
+                    "name" : "libsass",
+                    "buildsystem" : "meson",
+                    "cleanup" : [
+                        "*",
```
Comment (hfiguiere) [inline com.adrienplazas.Metronome.json:58]:
```suggestion
                        "*"
```

Comment (Kekun) [reply com.adrienplazas.Metronome.json]: /me shakes a fist in JSON's general direction.

**Comment by hfiguiere** [comment]
bot, build com.adrienplazas.Metronome

**Pull request closed without merging**